src/main/java/se/sundsvall/supportmanagement/service/RevisionService.java

```diff
@@ -0,0 +1,80 @@
+package se.sundsvall.supportmanagement.service;
+
+import static com.jayway.jsonpath.Configuration.defaultConfiguration;
+import static com.jayway.jsonpath.Option.SUPPRESS_EXCEPTIONS;
+import static org.apache.commons.lang3.ObjectUtils.anyNull;
+import static se.sundsvall.supportmanagement.service.mapper.RevisionMapper.toRevisionEntity;
+import static se.sundsvall.supportmanagement.service.mapper.RevisionMapper.toSerializedSnapshot;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.JsonPath;
+
+import se.sundsvall.supportmanagement.integration.db.RevisionRepository;
+import se.sundsvall.supportmanagement.integration.db.model.ErrandEntity;
+
+@Service
+public class RevisionService {
+	private static final String ERROR_MESSAGE = "An error occured when comparing previous and current entity as json";
+	private static final List<String> EXCLUDED_ATTRIBUTES = List.of("$.created", "$.modified", "$.touched", "$..stakeholders[*].id", "$..attachments[*].id");
+	private static final Configuration JSONPATH_CONFIG = defaultConfiguration().addOptions(SUPPRESS_EXCEPTIONS);
+	private static final Logger LOG = LoggerFactory.getLogger(RevisionService.class);
+
+	private final ObjectMapper objectMapper;
+
+	@Autowired
+	private RevisionRepository revisionRepository;
+
+	public RevisionService() {
+		objectMapper = new ObjectMapper();
+	}
+
+	/**
+	 * Create a new revision.
+	 *
+	 * A new revision will be created if:
+	 * - the last revisions serialized-snapshot differs from the current (i.e. provided) entity.
+	 * - no previous revisions exist for the provided entity.
+	 *
+	 * @param entity the entity that will have a new revision.
+	 */
+	public void createRevision(ErrandEntity entity) {
```

Comment:
Should we use historicReference (or whatever it was called) in eventHistoryService?
If so, maybe return the id of the created revision?

I suppose this id will be added in the eventService-call (as the historic reference)? 

Thoughts?